src/engine/Operation.h

```diff
@@ -446,6 +446,10 @@ class Operation {
       RuntimeInformation::Status status =
           RuntimeInformation::Status::optimizedOut);
 
+  // Return true if the given column originates from the loaded knowledge graph.
+  // This is used to skip potentially expensive checks.
```

Comment:
Be more explicit, e.g. 

"Return true if all values that the `variable` will be bound to by this expression are guaranteed to be contained in the underlying knowledge graph. This is e.g. true for results of `IndexScan`s, but not for `VALUES` clauses or expression results. This information is used to skip potentially expensive checks in the `TransitivePath` implementation."

Also: say something about UNDEF values (are they okay here or not? probably yes, but that should be documented)